### PR TITLE
#301: errors 

### PR DESCRIPTION
**Pull request by hani-ibrahim**
- working on #301 
- enhancing logs
- enhancing loading indicator look & feel

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/Utils/AtlasResult.swift [hunk 1]
```diff
@@ -6,38 +6,38 @@ import AtlasSDK
...
         }
     }
 
-    private func displayError(error: ErrorType, userMessage: UserMessage) {
-        guard let atlasError = error as? AtlasErrorType else {
+    private func displayError(error: ErrorType) {
+        guard let userPresentable = error as? UserPresentable else {
             // TODO: Need to check for network errors and other types
-            userMessage.generalError()
+            UserMessage.unclasifiedError(error)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/AtlasResult.swift]: It's probably my typo, but one "s" missed: unclassified

**Comment by bonkey** [comment]
Hm, whole `AtlasDemoUITests` suite failed?

**Comment by bonkey** [comment]
👍

**Comment by bonkey** [comment]
I'll try to rebuild without cache

**Comment by hani-ibrahim** [comment]
yeah tests are failing because of many issues 
working on it

**Comment by bonkey** [comment]
And travis tells `UserInfo={NSLocalizedDescription=Failed to get main window after 30 retries: kAXErrorServerNotFound (see <rdar://problem/15530121>)}`

**Comment by hani-ibrahim** [comment]
👍

**Comment by hani-ibrahim** [comment]
one more 👍

**Comment by bonkey** [comment]
👍

**Pull request merged**